app/main.py

```diff
@@ -1,3 +1,65 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", float, int]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
+            return Distance(self.km + other)
+
+    def __radd__(self, other: Union[float, int]) -> "Distance":
+        return self.__add__(other)
+
+    def __iadd__(self, other: Union["Distance", float, int]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: Union[float, int]) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Union[float, int]) -> "Distance":
+        return Distance(round(self.km / other, 2))
```

Comment:
`other` can not be of type `Distance` in this method

Reply:
what exactly do i need to change in the code?

Reply:
All correct for now ) Never mind , you did well 👍